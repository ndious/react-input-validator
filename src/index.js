import Text from './text'
import Email from './email';
import Password from './password'

export default {
  Text,
  Email,
  Password,
}
