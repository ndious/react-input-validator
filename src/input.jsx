import React from 'react';

export const Input = ({
  type, name, id,
  validator, required,
  onBlur, onChange,
  ...props
}) => {

  const test = target => {
    const isEmpty = target.value.length === 0
    let isValid
    if (typeof validator === 'function') {
      isValid = validator(target.value)
    } else {
      isValid = validator.test(target.value)
    }
    const onError = required ? (isEmpty || !isValid) : !(isValid || isEmpty)

    return {
      ...target,
      isEmpty,
      isValid,
      onError,
    }
  }

  const handleChange = ev => {
    console.log('there is change', onChange)
    if (onChange) {
      return onChange({
        ...ev,
        target: test(ev.target)
      })
    }
  }

  const handleBlur = ev => {
    if (onBlur) {
      return onBlur({
        ...ev,
        target: test(ev.target)
      })
    }
  }

  return (
    <input
      type={type}
      name={name || id}
      id={id || name}
      onChange={handleChange}
      onBlur={handleBlur}
      {...props}
      required
    />
  )
}
