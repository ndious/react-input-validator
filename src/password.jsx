import React from 'react'
import Input from './input'

const Password = ({min, max, upper, lower, special, number, ...props}) => {
  let validator = '^'
  validator += lower ? '(?=.*[a-z])' : ''
  validator += upper ? '(?=.*[A-Z])' : ''
  validator += number ? '(?=.*\\d)' : ''
  validator += special ? '(?=.*[$@$!%*?&])' : ''
  validator += `[\\w\\d$@$!%*?&]{${min || 0},${max || ''}}$`

  return (
    <Input
      type="password"
      validator={new RegExp(validator)}
      {...props}
    />
  )
}

export default Password
