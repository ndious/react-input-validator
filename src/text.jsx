import React from 'react'
import {Input }from './input'

const Text = ({validate, ...props}) => (
  <Input type="text" validator={validate || /.*/} {...props} />
)

export default Text
