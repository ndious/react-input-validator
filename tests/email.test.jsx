import React from 'react'
import { shallow } from 'enzyme'

import Email from '../src/email'
import { Input } from '../src/input'

it('should render a input', () => {
    const wrapper = shallow(<Email />)
    expect(wrapper).toMatchSnapshot()
});
