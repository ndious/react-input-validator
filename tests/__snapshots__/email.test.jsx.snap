// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a input 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Input
    type="email"
    validator={/\\(\\?:\\[a-z0-9!#\\$%&'\\*\\+\\\\/=\\?\\^_\`\\{\\|\\}~-\\]\\+\\(\\?:\\\\\\.\\[a-z0-9!#\\$%&'\\*\\+\\\\/=\\?\\^_\`\\{\\|\\}~-\\]\\+\\)\\*\\|"\\(\\?:\\[\\\\x01-\\\\x08\\\\x0b\\\\x0c\\\\x0e-\\\\x1f\\\\x21\\\\x23-\\\\x5b\\\\x5d-\\\\x7f\\]\\|\\\\\\\\\\[\\\\x01-\\\\x09\\\\x0b\\\\x0c\\\\x0e-\\\\x7f\\]\\)\\*"\\)@\\(\\?:\\(\\?:\\[a-z0-9\\]\\(\\?:\\[a-z0-9-\\]\\*\\[a-z0-9\\]\\)\\?\\\\\\.\\)\\+\\[a-z0-9\\]\\(\\?:\\[a-z0-9-\\]\\*\\[a-z0-9\\]\\)\\?\\|\\\\\\[\\(\\?:\\(\\?:25\\[0-5\\]\\|2\\[0-4\\]\\[0-9\\]\\|\\[01\\]\\?\\[0-9\\]\\[0-9\\]\\?\\)\\\\\\.\\)\\{3\\}\\(\\?:25\\[0-5\\]\\|2\\[0-4\\]\\[0-9\\]\\|\\[01\\]\\?\\[0-9\\]\\[0-9\\]\\?\\|\\[a-z0-9-\\]\\*\\[a-z0-9\\]:\\(\\?:\\[\\\\x01-\\\\x08\\\\x0b\\\\x0c\\\\x0e-\\\\x1f\\\\x21-\\\\x5a\\\\x53-\\\\x7f\\]\\|\\\\\\\\\\[\\\\x01-\\\\x09\\\\x0b\\\\x0c\\\\x0e-\\\\x7f\\]\\)\\+\\)\\\\\\]\\)/}
/>,
  "nodes": Array [
    <Input
      type="email"
      validator={/\\(\\?:\\[a-z0-9!#\\$%&'\\*\\+\\\\/=\\?\\^_\`\\{\\|\\}~-\\]\\+\\(\\?:\\\\\\.\\[a-z0-9!#\\$%&'\\*\\+\\\\/=\\?\\^_\`\\{\\|\\}~-\\]\\+\\)\\*\\|"\\(\\?:\\[\\\\x01-\\\\x08\\\\x0b\\\\x0c\\\\x0e-\\\\x1f\\\\x21\\\\x23-\\\\x5b\\\\x5d-\\\\x7f\\]\\|\\\\\\\\\\[\\\\x01-\\\\x09\\\\x0b\\\\x0c\\\\x0e-\\\\x7f\\]\\)\\*"\\)@\\(\\?:\\(\\?:\\[a-z0-9\\]\\(\\?:\\[a-z0-9-\\]\\*\\[a-z0-9\\]\\)\\?\\\\\\.\\)\\+\\[a-z0-9\\]\\(\\?:\\[a-z0-9-\\]\\*\\[a-z0-9\\]\\)\\?\\|\\\\\\[\\(\\?:\\(\\?:25\\[0-5\\]\\|2\\[0-4\\]\\[0-9\\]\\|\\[01\\]\\?\\[0-9\\]\\[0-9\\]\\?\\)\\\\\\.\\)\\{3\\}\\(\\?:25\\[0-5\\]\\|2\\[0-4\\]\\[0-9\\]\\|\\[01\\]\\?\\[0-9\\]\\[0-9\\]\\?\\|\\[a-z0-9-\\]\\*\\[a-z0-9\\]:\\(\\?:\\[\\\\x01-\\\\x08\\\\x0b\\\\x0c\\\\x0e-\\\\x1f\\\\x21-\\\\x5a\\\\x53-\\\\x7f\\]\\|\\\\\\\\\\[\\\\x01-\\\\x09\\\\x0b\\\\x0c\\\\x0e-\\\\x7f\\]\\)\\+\\)\\\\\\]\\)/}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Email />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Input
          type="email"
          validator={/\\(\\?:\\[a-z0-9!#\\$%&'\\*\\+\\\\/=\\?\\^_\`\\{\\|\\}~-\\]\\+\\(\\?:\\\\\\.\\[a-z0-9!#\\$%&'\\*\\+\\\\/=\\?\\^_\`\\{\\|\\}~-\\]\\+\\)\\*\\|"\\(\\?:\\[\\\\x01-\\\\x08\\\\x0b\\\\x0c\\\\x0e-\\\\x1f\\\\x21\\\\x23-\\\\x5b\\\\x5d-\\\\x7f\\]\\|\\\\\\\\\\[\\\\x01-\\\\x09\\\\x0b\\\\x0c\\\\x0e-\\\\x7f\\]\\)\\*"\\)@\\(\\?:\\(\\?:\\[a-z0-9\\]\\(\\?:\\[a-z0-9-\\]\\*\\[a-z0-9\\]\\)\\?\\\\\\.\\)\\+\\[a-z0-9\\]\\(\\?:\\[a-z0-9-\\]\\*\\[a-z0-9\\]\\)\\?\\|\\\\\\[\\(\\?:\\(\\?:25\\[0-5\\]\\|2\\[0-4\\]\\[0-9\\]\\|\\[01\\]\\?\\[0-9\\]\\[0-9\\]\\?\\)\\\\\\.\\)\\{3\\}\\(\\?:25\\[0-5\\]\\|2\\[0-4\\]\\[0-9\\]\\|\\[01\\]\\?\\[0-9\\]\\[0-9\\]\\?\\|\\[a-z0-9-\\]\\*\\[a-z0-9\\]:\\(\\?:\\[\\\\x01-\\\\x08\\\\x0b\\\\x0c\\\\x0e-\\\\x1f\\\\x21-\\\\x5a\\\\x53-\\\\x7f\\]\\|\\\\\\\\\\[\\\\x01-\\\\x09\\\\x0b\\\\x0c\\\\x0e-\\\\x7f\\]\\)\\+\\)\\\\\\]\\)/}
/>,
        "_debugID": 2,
        "_renderedOutput": <Input
          type="email"
          validator={/\\(\\?:\\[a-z0-9!#\\$%&'\\*\\+\\\\/=\\?\\^_\`\\{\\|\\}~-\\]\\+\\(\\?:\\\\\\.\\[a-z0-9!#\\$%&'\\*\\+\\\\/=\\?\\^_\`\\{\\|\\}~-\\]\\+\\)\\*\\|"\\(\\?:\\[\\\\x01-\\\\x08\\\\x0b\\\\x0c\\\\x0e-\\\\x1f\\\\x21\\\\x23-\\\\x5b\\\\x5d-\\\\x7f\\]\\|\\\\\\\\\\[\\\\x01-\\\\x09\\\\x0b\\\\x0c\\\\x0e-\\\\x7f\\]\\)\\*"\\)@\\(\\?:\\(\\?:\\[a-z0-9\\]\\(\\?:\\[a-z0-9-\\]\\*\\[a-z0-9\\]\\)\\?\\\\\\.\\)\\+\\[a-z0-9\\]\\(\\?:\\[a-z0-9-\\]\\*\\[a-z0-9\\]\\)\\?\\|\\\\\\[\\(\\?:\\(\\?:25\\[0-5\\]\\|2\\[0-4\\]\\[0-9\\]\\|\\[01\\]\\?\\[0-9\\]\\[0-9\\]\\?\\)\\\\\\.\\)\\{3\\}\\(\\?:25\\[0-5\\]\\|2\\[0-4\\]\\[0-9\\]\\|\\[01\\]\\?\\[0-9\\]\\[0-9\\]\\?\\|\\[a-z0-9-\\]\\*\\[a-z0-9\\]:\\(\\?:\\[\\\\x01-\\\\x08\\\\x0b\\\\x0c\\\\x0e-\\\\x1f\\\\x21-\\\\x5a\\\\x53-\\\\x7f\\]\\|\\\\\\\\\\[\\\\x01-\\\\x09\\\\x0b\\\\x0c\\\\x0e-\\\\x7f\\]\\)\\+\\)\\\\\\]\\)/}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Email />,
}
`;
